pkg/apis/workflow/v1alpha1/artifact_gc.go

```diff
@@ -0,0 +1,52 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +kubebuilder:resource:shortName=agw
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:subresource:status
+type ArtifactGCTask struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata" protobuf:"bytes,1,opt,name=metadata"`
+	Spec              ArtifactGCSpec    `json:"spec" protobuf:"bytes,2,opt,name=spec"`
+	Status            ArtifactGCSStatus `json:"status,omitempty" protobuf:"bytes,3,opt,name=status"`
+}
+
+type ArtifactGCSpec struct {
+	ArtifactsByNode map[string]ArtifactNodeSpec `json:"artifactsByNode,omitempty" protobuf:"bytes,1,rep,name=artifactsByNode"`
+}
+
+type ArtifactNodeSpec struct {
+	ArchiveLocation *ArtifactLocation       `json:"archiveLocation,omitempty" protobuf:"bytes,1,opt,name=archiveLocation"`
+	Artifacts       map[string]ArtifactList `json:"artifacts,omitempty" protobuf:"bytes,2,rep,name=artifacts"`
+}
+
+type ArtifactList struct {
```

Comment:
Do you really need Artifactlist?

Reply:
I initially tried not using it and just using "Artifacts" there but I guess because "Artifacts" doesn't have a json definition associated with it, the .proto generator turned it into a map<string, Artifact>.

Reply:
I did try making json "inline" but that seemed to fail compilation as well so I went back to this version.

Reply:
Actually, I realized I had a bug there and that instead of having a "map[string]ArtifactList" I actually do want a "map[string]Artifact" anyway :) so I can remove that extra struct.